File: 474. Ones and Zeroes/Java/Ones and Zeroes.java
public class Solution {
    public int findMaxForm(String[] strs, int m, int n) {
        int[][] dp = new int[m + 1][n + 1];
        for (String str : strs) {
            int zeros = 0;
            int ones = 0;
            for (char letter : str.toCharArray()) {
                if (letter == '0') {
                    zeros++;
                } else {
                    ones++;
                }
            }
            for (int i = m; i >= 0; i--) {
                for (int j = n; j >= 0; j--) {
                    if (i >= zeros && j >= ones) {
                        dp[i][j] = Math.max(dp[i][j], dp[i - zeros][j - ones] + 1);
                    }
                }
            }
        }
        return dp[m][n];
    }
}

File: 474. Ones and Zeroes/Python/Ones and Zeroes.py
class Solution(object):
    def findMaxForm(self, strs, m, n):
        """
        :type strs: List[str]
        :type m: int
        :type n: int
        :rtype: int
        """
        dp = [[0] * (n + 1) for i in range(m + 1)]
        for str in strs:
            zeros = 0
            ones = 0
            for letter in str:
                if letter == '0':
                    zeros += 1
                else:
                    ones += 1
            for i in range(m, -1, -1):
                for j in range(n, -1, -1):
                    if i >= zeros and j >= ones:
                        dp[i][j] = max(dp[i][j], dp[i - zeros][j - ones] + 1)
        return dp[m][n]
